goagen/gen_swagger/swagger.go

```diff
@@ -347,6 +354,24 @@ func New(api *design.APIDefinition) (*Swagger, error) {
 	return s, nil
 }
 
+func tagsFromDefinition(mdatas []design.MetadataDefinition) (tags []Tag, tagNames []string, err error) {
```

Comment:
Maybe create two functions instead of one that can return both tags and names? i.e. a `tagNamesFromDefinitions` and a `tagsFromDefinition`. The former takes a slice of `design.MetadataDefinition` (like you have it now) but the latter just a single `*design.MetadataDefinition`.